src/main/java/life/qbic/data/processing/registration/ValidationException.java

```diff
@@ -0,0 +1,26 @@
+package life.qbic.data.processing.registration;
+
+/**
+ * <b><class short description - 1 Line!></b>
+ *
+ * <p><More detailed description - When to use, what it solves, etc.></p>
+ *
+ * @since <version tag>
+ */
+public class ValidationException extends RuntimeException {
+
+  private final ErrorCode errorCode;
+
+  public ValidationException(ErrorCode errorCode) {
+    this.errorCode = errorCode;
+  }
+
+  public ValidationException(String message, ErrorCode errorCode) {
+    super(message);
+    this.errorCode = errorCode;
+  }
+
+  public ErrorCode errorCode() {
+    return errorCode;
+  }
+}
```

Comment:
Question: Should the ValidationException be chainable so the user gets a report of all invalid errors instead of retrying time after time? 🤔